examples/2d-measure-accuracy.py

```diff
@@ -23,32 +23,50 @@ def parse_args():
 def no_regularizer(amt):
     return None
  
+def l2_distance(a,b):
+    return tf.square(a-b)
+
+def l1_distance(a,b):
+    return a-b
+
 def custom_discriminator_config():
-    return { 
-            'create': custom_discriminator 
-    }
+    selector = hc.Selector()
+    selector.set('create', custom_discriminator)
+    selector.set('distance',[l2_distance, l1_distance])
+    return selector.random_config()
 
 def custom_generator_config():
     return { 
             'create': custom_generator
     }
+def custom_encoder_config():
+    return { 
+            'create': custom_encoder
+    }
+
 
 
 def custom_discriminator(gan, config, x, g, xs, gs, prefix='d_'):
     net = tf.concat(axis=0, values=[x,g])
+    original = net
     net = linear(net, 128, scope=prefix+'linone')
-    net = tf.nn.crelu(net)
-    net = linear(net, 128, scope=prefix+'linend')
+    net = tf.nn.relu(net)
+    net = linear(net, 2, scope=prefix+'linend')
+    
+    # works.  hyperparam? 
+    net = config.distance(original,net)
```

Comment:
This only makes sense if you want to build an autoencoder.  I wonder how we can refactor this?